zebrad/src/components/inbound.rs

```diff
@@ -352,10 +352,16 @@ impl Service<zn::Request> for Inbound {
                     })
                 .boxed()
             }
-            zn::Request::PushTransaction(_transaction) => {
-                debug!("ignoring unimplemented request");
-                // TODO: send to Tx Download & Verify Stream
-                // https://github.com/ZcashFoundation/zebra/issues/2692
+            zn::Request::PushTransaction(transaction) => {
+                if let Setup::Initialized { tx_downloads, .. } = &mut self.network_setup {
+                    // TODO: check if we're close to the tip before proceeding?
+                    // what do we do if it's not?
```

Comment:
It might be easier to do the mempool activation check inside the mempool service or download stream, so we don't have to do them in each caller. (Otherwise, we'd have the same code in the mempool crawler, inbound service, and client.)

Reply:
I agree! I believe this will end up being solved in https://github.com/ZcashFoundation/zebra/issues/2728